int64 id
---
uint8 ret